### PR TITLE
Issue/1625 implement element closest

### LINKED ISSUE #1625: NoSuchElement in Android Appium test after updating to Gutenberg master
Trying to point to Gutenberg master hash https://github.com/WordPress/gutenberg/commit/d384e167526d3eec152a707fa7592debc56ec648, the Android Appium tests fail with:

```
 FAIL  __device-tests__/gutenberg-editor-paste.test.js (158.51s)
  Gutenberg Editor paste tests
    ✓ copies plain text from one paragraph block and pastes in another (47412ms)
    ✕ copies styled text from one paragraph block and pastes in another (65502ms)

  ● Gutenberg Editor paste tests › copies styled text from one paragraph block and pastes in another

    [elementByXPath("//*[@content-desc=\"html-view-content, \"]")] Error response status: 7, , NoSuchElement - An element could not be located on the page using the given search parameters. Selenium error: An element could not be located on the page using the given search parameters.

      at Object.<anonymous>.exports.newError (node_modules/wd/lib/utils.js:152:13)
      at node_modules/wd/lib/callbacks.js:94:19
      at node_modules/wd/lib/webdriver.js:194:5
      at Request._callback (node_modules/wd/lib/http-utils.js:89:7)
      at Request.self.callback (node_modules/request/request.js:185:22)
      at Request.<anonymous> (node_modules/request/request.js:1161:10)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:1083:12)
```

@cameronvoell and @etoledom have pinnpointed the breaking Gutenberg change to the [Fix crash on mobile when pasting formatted text](https://github.com/WordPress/gutenberg/pull/18740) PR and the root cause to missing the `Element.closest()` function in the `jsdom-jscore` used in gutenberg-mobile.

### PR DESCRIPTION
**Pull request by hypest**
Fixes #1625

Adds to the local jsdom patches the implementation of `Element.closest()` as seen at https://github.com/jsdom/jsdom/issues/1555#issuecomment-264288212.

Even though jsdom itself has had that function added already, the jsdom-jscore fork we are using is not up-to-date to use it. Updating the fork is far from trivial so, resorting to a local patch instead.

To test:
* Android Appium tests on CI should pass

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: src/jsdom-patches.js [hunk 1]
```diff
@@ -149,6 +149,16 @@ Node.prototype.insertBefore = function( /* Node */ newChild, /* Node*/ refChild
...
 
+Element.prototype.closest = function( selector ) {
+	let el = this;
+	while ( el ) {
+		if ( el.matches( selector ) ) {
+			return el;
+		}
+		el = el.parentElement;
+	}
+};
```
[the remark this answers is not in the document]
Comment (etoledom) [reply src/jsdom-patches.js]: Nice catch @mkevins !

**Comment by hypest** [comment]
> is the gutenberg reference update related?

Not really. There are no Gutenberg side changes that are relevant. I included an update of the ref for making sure this fix does indeed fix the tests that only get broken if you update to a recent Gutenberg hash. Other than that, this can be considered a typical Gutenberg "bump" to master to bring in the latest.

Would you prefer to just leave out the Gutenberg hash update @mkevins , now that we now the tests pass?

File: src/jsdom-patches.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (hypest) [reply src/jsdom-patches.js]: Makes sense! Done with 919526cf.

**Comment by hypest** [comment]
Ready for another pass!

**Comment by cameronvoell** [comment]
> Ready for another pass!

I verified that we no longer crash, and we also correctly identify the pre tag using the newly added element.closest function, and we correctly return null when the <pre> tag is not there. Thanks for adding this @hypest 

Since we already have two approvals, I think we can go ahead and merge, unless @mkevins has any other concerns on the newly added code?

**Comment by mkevins** [comment]
> > is the gutenberg reference update related?
> 
> Not really. There are no Gutenberg side changes that are relevant. I included an update of the ref for making sure this fix does indeed fix the tests that only get broken if you update to a recent Gutenberg hash. Other than that, this can be considered a typical Gutenberg "bump" to master to bring in the latest.
> 
> Would you prefer to just leave out the Gutenberg hash update @mkevins , now that we know the tests pass?

No need I just wanted to make sure I wasn't missing anything. Thanks for the explanation, makes sense!

**Comment by mkevins** [comment]
> > Ready for another pass!
> 
> Thanks for adding this @hypest
> 
> Since we already have three approvals, I think we can go ahead and merge, unless @mkevins has any other concerns on the newly added code?

No concerns here. LGTM! :shipit:

**Pull request merged**